src/app/logic/Herd.ts

```diff
@@ -128,35 +132,60 @@ export class Herd {
    */
   // TODO: Check parameteres type. Create classes for protectors and predators if needed.
   // TODO: Modify to use config? Define at refactor
-  cullAnimals(attackingAnimal: Fox | Wolf): void {
-    switch (attackingAnimal.theName) {
-      case AnimalNames.FOX: {
-        const hasSmallDog =
-          this.getAnimalNumber(AnimalNames.SMALL_DOG) > 0;
-        if (!hasSmallDog) {
-          attackingAnimal.attackHerd();
-          this.cullAllAnimalsOfOneType(AnimalNames.RABBIT);
-          return;
-        }
-        this.removeAnimalsFromHerd(AnimalNames.SMALL_DOG, 1);
-        // (this.animals[5][0] as SmallDog).protectHerd();
-        break;
+  cullAnimals(
+    //  TODO: REMOVE FOX | WOLF AFTER BREED REFACTOR
+    attackingAnimal: Predator | Fox | Wolf,
+    mode?: GameModes,
+  ): void {
+    if (attackingAnimal instanceof Predator) {
+      // TODO: ADD SWITCH FOR GAME MODE
+      const animalsToCull = attackingAnimal.kills;
+      const protector = attackingAnimal.isChasedAwayBy;
+      const protectorIndex = this.findAnimalTupleIndex(protector);
+      const hasProtector = this.getAnimalNumber(protector) > 0;
+      console.log(protector);
+      console.log(animalsToCull);
+      console.log(JSON.parse(JSON.stringify(this.animals)));
+      if (!hasProtector) {
+        console.log(hasProtector, ' no protectos');
+        this.cullAllAnimalsOfGivenTypes(animalsToCull);
+        attackingAnimal.attackHerd();
       }
-      case AnimalNames.WOLF: {
-        const hasBigDog =
-          this.getAnimalNumber(AnimalNames.BIG_DOG) > 0;
-        if (!hasBigDog) {
-          attackingAnimal.attackHerd();
-          this.cullAllAnimalsOfGivenTypes([
-            AnimalNames.COW,
-            AnimalNames.PIG,
-            AnimalNames.RABBIT,
-            AnimalNames.SHEEP,
-          ]);
-          return;
+      this.removeAnimalsFromHerd(protector, 1);
+      console.log(JSON.parse(JSON.stringify(this.animals)));
+      // TODO: ADD METHOD TO PROTECTORS
+      // this.animals[protectorIndex][0].protectHerd();
+    } else {
+      // TODO: REMOVE AFTER BREED PROCESSOR REFACTOR
+      switch (attackingAnimal.theName) {
+        case AnimalNames.FOX: {
+          const hasSmallDog =
+            this.getAnimalNumber(AnimalNames.SMALL_DOG) > 0;
```

Comment:
I guess you often have to check if an animal exists on farm, a separate method might be useful.